src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmailController.php

```diff
@@ -97,7 +97,7 @@ public function indexAction(Request $request, EmailLogsFilter $filters)
     public function searchAction(Request $request)
     {
         $definitionFactory = $this->get('prestashop.core.grid.definition.factory.email_logs');
-        $emailLogsDefinition = $definitionFactory->create();
+        $emailLogsDefinition = $definitionFactory->getDefinition();
```

Comment:
not really, it's because we're doing object construction in controllers when we shouldn't do it: we should rely on the grid directly instead of rebuilding the form from grid definition

Reply:
> it's because we're doing object construction in controllers when we shouldn't do

agree, but if we were to create whole grid, we would need to do unnecessary SQL queries even though db data is not used here, only filter form is important. :/

At the end:

> “There are no solutions. There are only trade-offs.”
> ― Thomas Sowell